### ISSUE TITLE
Screen reader fails to announce the updated state information which is visually indicated on the button when the user navigates to the 'Selection Support Pressable' button.

### ISSUE THREAD

**Issue by Nitin-100** [microsoft/react-native-gallery#658]
Build Details:
OS Version: 24H2 
OS Build: 26100.5074
App: React Native Gallery Preview
AT Used: Narrator

Pre-Requisite:
Turn on Narrator (Ctrl + Win + Enter).

Repro-Steps:
Launch the React Native Gallery-Preview app.
Navigate to the 'Expand menu' button and invoke it.
Navigate to 'All samples' button and invoke it.
Navigate to the 'Virtualized list' item and invoke it.
Navigate to the 'Selection support' button and invoke it.
Now listen to the screen reader announcement.
Actual Result:
Narrator fails to announce the updated state information which is visually indicated on the button when the user navigates to the 'Selection Support Pressable' button, and it is simply announcing as "selection support pressable, button, click me to change selection support".
 
Expected Result:
When the user navigates to the 'Selection Support Pressable' button, narrator should announce the updated state information which is visually indicated on the button like "selection support pressable, multiple, button, click me to change selection support".
 
User Impact:
User with visual impairments who rely on screen reader are unable to perceive the current selection mode because it is not announced by Narrator. This leads to confusion about the state of the control, prevents users from understanding what action has occurred, and limits their ability to interact with the UI effectively.

MAS Reference:
[MAS 4.2.1 – Object Information](https://aka.ms/MAS4.2.1)

Trap ID:
[1.4 - Uncomprehended Element](https://aka.ms/UncomprehendedElement)

“Have feedback on bugs logged, please tag bug as “A11yRCA” and add your feedback in the comment section.”
